src/GameServer.js

```diff
@@ -623,8 +651,12 @@ GameServer.prototype.updateMoveEngine = function() {
     len = this.nonPlayerNodes.length;
```

Comment:
Everything having their own timer is a horrible idea. It will just complicate everything. Instead of this, you should make a "master timer" and measure ticks using that, where the MT goes up in a constant rate as time goes and every single "subtimer" refers to that single timer and adjusts to it. Although more complicated, it will pay off with performance. The goal is to make the code as simple and flawless as possible, and ogar is terribly coded. You need to also have a buffer variable for each client if you want it to work smoothly, so that the asynchronous processes can work well with other processes which is better run synchronously.